### PR TITLE
Tweak validation of backuptarget URL.

### PR DESCRIPTION
**Pull request by james-munson**
https://github.com/longhorn/longhorn/issues/6608

Accept NFS mount options as a query appended to the backup target URL.
Confine the check for bogus characters "," and "$" to the URL path, not the whole string.

Can be separable from the vendor update to be done after backupstore is merged.

### CODE REVIEW

**Comment by innobead** [comment]
@james-munson Please resolve the conflict.

File: types/setting.go [hunk 1]
```diff
@@ -1203,15 +1203,15 @@ func ValidateSetting(name, value string) (err error) {
...
+		// Check whether have $ or , have been set in BackupTarget path
 		regStr := `[\$\,]`
 		if u.Scheme == "cifs" {
 			// The $ in SMB/CIFS URIs means that the share is hidden.
 			regStr = `[\,]`
 		}
 
 		reg := regexp.MustCompile(regStr)
-		findStr := reg.FindAllString(value, -1)
+		findStr := reg.FindAllString(u.Path, -1)
```
Comment (PhanLe1010) [inline types/setting.go:1214]: LGTM. Nice catch!

**Review by ejweber — APPROVED** [review]

**Review by ejweber — APPROVED** [review]
LGTM. I'm assuming the previously included dependency updates will be in a different PR once the backupstore PR is ready for merge.

**Comment by james-munson** [comment]
@mergify backport v1.5.x

**Pull request merged**